src/main/java/jsr223/docker/file/DockerFileScriptEngine.java

```diff
@@ -0,0 +1,164 @@
+ * or a different license than the AGPL.
+ */
+package jsr223.docker.file;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.Reader;
+import java.io.StringWriter;
+import java.util.Map;
+
+import javax.script.*;
+
+import jsr223.docker.compose.bindings.MapBindingsAdder;
+import jsr223.docker.compose.bindings.StringBindingsAdder;
+import jsr223.docker.compose.file.write.ConfigurationFileWriter;
+import jsr223.docker.compose.utils.Log4jConfigurationLoader;
+import jsr223.docker.compose.utils.ScriptContextBindingsExtractor;
+import jsr223.docker.compose.yaml.VariablesReplacer;
+import jsr223.docker.file.DockerFileCommandCreator;
+import jsr223.docker.file.DockerFileScriptEngine;
+import jsr223.docker.file.DockerFileScriptEngineFactory;
+import lombok.extern.log4j.Log4j;
+import processbuilder.SingletonProcessBuilderFactory;
+import processbuilder.utils.ProcessBuilderUtilities;
+
+
+@Log4j
+public class DockerFileScriptEngine extends AbstractScriptEngine {
+
+    private static final String DOCKER_HOST_PROPERTY_NAME = "DOCKER_HOST";
+
+    private ProcessBuilderUtilities processBuilderUtilities = new ProcessBuilderUtilities();
+
+    private VariablesReplacer variablesReplacer = new VariablesReplacer();
+
+    private ConfigurationFileWriter configurationFileWriter = new ConfigurationFileWriter();
+
+    private StringBindingsAdder stringBindingsAdder = new StringBindingsAdder(new MapBindingsAdder());
+
+    private DockerFileCommandCreator dockerFileCommandCreator = new DockerFileCommandCreator();
+
+    private Log4jConfigurationLoader log4jConfigurationLoader = new Log4jConfigurationLoader();
+
+    public DockerFileScriptEngine() {
+        // This is the entry-point of the script engine
+        log4jConfigurationLoader.loadLog4jConfiguration();
+    }
+
+    @Override
+    public Object eval(String script, ScriptContext context) throws ScriptException {
+
+        // Create docker file command - a simple docker build command 
+        String[] dockerFileCommand = dockerFileCommandCreator.createDockerFileExecutionCommand();
+
+        // Create a process builder
+        ProcessBuilder processBuilder = SingletonProcessBuilderFactory.getInstance()
+                                                                      .getProcessBuilder(dockerFileCommand);
+
+        // Use process builder environment and fill it with environment variables
+        Map<String, String> variablesMap = processBuilder.environment();
+
+        // Add string bindings as environment variables
+        stringBindingsAdder.addBindingToStringMap(context.getBindings(ScriptContext.ENGINE_SCOPE), variablesMap);
+
+        // Add DOCKER_HOST variable to execution environment
+        variablesMap.put(DOCKER_HOST_PROPERTY_NAME, DockerFilePropertyLoader.getInstance().getDockerHost());
+
+        // Replace variables in configuration file
+        String scriptReplacedVariables = variablesReplacer.replaceVariables(script, variablesMap);
```

Comment:
I think we don't need to replace variables inside the Dockerfile. We should use bash and the --build-arg options to provide variables in a docker fashion.